--- a/net/minecraft/world/level/block/EndPortalFrameBlock.java
+++ b/net/minecraft/world/level/block/EndPortalFrameBlock.java
@@ -27,7 +_,7 @@
     public static final EnumProperty<Direction> FACING = HorizontalDirectionalBlock.FACING;
     public static final BooleanProperty HAS_EYE = BlockStateProperties.EYE;
     protected static final VoxelShape BASE_SHAPE = Block.box(0.0, 0.0, 0.0, 16.0, 13.0, 16.0);
-    protected static final VoxelShape EYE_SHAPE = Block.box(4.0, 13.0, 4.0, 12.0, 16.0, 12.0);
+    protected static final VoxelShape EYE_SHAPE = Block.box(5.0, 13.0, 5.0, 11.0, 16.0, 11.0);
     protected static final VoxelShape FULL_SHAPE = Shapes.or(BASE_SHAPE, EYE_SHAPE);
     private static BlockPattern portalShape;
 
