From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Melody Belle <melody@meluhdy.dev>
Date: Sat, 4 Apr 2026 20:35:40 -0400
Subject: [PATCH] [PATCH] 1.12 Bed Hitboxes


diff --git a/net/minecraft/world/level/block/BedBlock.java b/net/minecraft/world/level/block/BedBlock.java
index 7bfc62120cae4c5e83cb0d86b759b7ffb2336a95..0441942e0882a58b1331e7e5cf86f076c470c124 100644
--- a/net/minecraft/world/level/block/BedBlock.java
+++ b/net/minecraft/world/level/block/BedBlock.java
@@ -54,11 +54,8 @@ public class BedBlock extends HorizontalDirectionalBlock implements EntityBlock
     );
     public static final EnumProperty<BedPart> PART = BlockStateProperties.BED_PART;
     public static final BooleanProperty OCCUPIED = BlockStateProperties.OCCUPIED;
-    private static final Map<Direction, VoxelShape> SHAPES = Util.make(() -> {
-        VoxelShape voxelShape = Block.box(0.0, 0.0, 0.0, 3.0, 3.0, 3.0);
-        VoxelShape voxelShape1 = Shapes.rotate(voxelShape, OctahedralGroup.BLOCK_ROT_Y_90);
-        return Shapes.rotateHorizontal(Shapes.or(Block.column(16.0, 3.0, 9.0), voxelShape, voxelShape1));
-    });
+    protected static final int HEIGHT = 9;
+    protected static final VoxelShape BASE = Block.box(0.0, 0.0, 0.0, 16.0, HEIGHT, 16.0);
     private final DyeColor color;
 
     @Override
@@ -244,7 +241,7 @@ public class BedBlock extends HorizontalDirectionalBlock implements EntityBlock
 
     @Override
     protected VoxelShape getShape(BlockState state, BlockGetter level, BlockPos pos, CollisionContext context) {
-        return SHAPES.get(getConnectedDirection(state).getOpposite());
+        return BASE;
     }
 
     public static Direction getConnectedDirection(BlockState state) {
