From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Melody Belle <melody@meluhdy.dev>
Date: Sat, 4 Apr 2026 20:40:20 -0400
Subject: [PATCH] [PATCH] 1.12 End Portal Frame Hitboxes


diff --git a/net/minecraft/world/level/block/EndPortalFrameBlock.java b/net/minecraft/world/level/block/EndPortalFrameBlock.java
index f932219476da6ddb04d9abbc53b6283888765083..1920ad4276fe97922b57fececc7b0bd1a2598204 100644
--- a/net/minecraft/world/level/block/EndPortalFrameBlock.java
+++ b/net/minecraft/world/level/block/EndPortalFrameBlock.java
@@ -27,8 +27,9 @@ public class EndPortalFrameBlock extends Block {
     public static final MapCodec<EndPortalFrameBlock> CODEC = simpleCodec(EndPortalFrameBlock::new);
     public static final EnumProperty<Direction> FACING = HorizontalDirectionalBlock.FACING;
     public static final BooleanProperty HAS_EYE = BlockStateProperties.EYE;
-    private static final VoxelShape SHAPE_EMPTY = Block.column(16.0, 0.0, 13.0);
-    private static final VoxelShape SHAPE_FULL = Shapes.or(SHAPE_EMPTY, Block.column(8.0, 13.0, 16.0));
+    protected static final VoxelShape BASE_SHAPE = Block.box(0.0, 0.0, 0.0, 16.0, 13.0, 16.0);
+    protected static final VoxelShape EYE_SHAPE = Block.box(5.0, 13.0, 5.0, 11.0, 16.0, 11.0);
+    protected static final VoxelShape FULL_SHAPE = Shapes.or(BASE_SHAPE, EYE_SHAPE);
     private static @Nullable BlockPattern portalShape;
 
     @Override
@@ -48,7 +49,7 @@ public class EndPortalFrameBlock extends Block {
 
     @Override
     protected VoxelShape getShape(BlockState state, BlockGetter level, BlockPos pos, CollisionContext context) {
-        return state.getValue(HAS_EYE) ? SHAPE_FULL : SHAPE_EMPTY;
+        return state.getValue(HAS_EYE) ? FULL_SHAPE : BASE_SHAPE;
     }
 
     @Override
