From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Melody Belle <melody@meluhdy.dev>
Date: Mon, 24 Aug 2026 22:56:03 -0400
Subject: [PATCH] [PATCH] Remove Secure Message Check


diff --git a/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index f936da2a1ffef5fd84d8d63be3575fd06ac9a7b2..fcabd33205ab2e5b84249a055117488559f14b18 100644
--- a/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -338,7 +338,7 @@ public class ServerGamePacketListenerImpl
         this.player = player;
         player.connection = this;
         player.getTextFilter().join();
-        this.signedMessageDecoder = SignedMessageChain.Decoder.unsigned(player.getUUID(), server::enforceSecureProfile);
+        this.signedMessageDecoder = SignedMessageChain.Decoder.unsigned(player.getUUID(), () -> false);
         this.chatMessageChain = new FutureChain(server.chatExecutor); // CraftBukkit - async chat
         this.tickEndEvent = new io.papermc.paper.event.packet.ClientTickEndEvent(player.getBukkitEntity()); // Paper - add client tick end event
         this.playerGameConnection = new io.papermc.paper.connection.PaperPlayerGameConnection(this); // Paper
