From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Melody Belle <melody@meluhdy.dev>
Date: Thu, 30 Apr 2026 17:20:48 -0400
Subject: [PATCH] [PATCH] Remove 1.21 Player physics


diff --git a/net/minecraft/world/entity/player/Player.java b/net/minecraft/world/entity/player/Player.java
index 7bf160cef8bd772d27d4a71f04d5d9861ef67984..371e396e951450dc670218029700c7df570151e9 100644
--- a/net/minecraft/world/entity/player/Player.java
+++ b/net/minecraft/world/entity/player/Player.java
@@ -245,7 +245,7 @@ public abstract class Player extends Avatar implements ContainerUser {
 
     @Override
     public void tick() {
-        this.noPhysics = this.isSpectator();
+        this.noPhysics = true;
         if (this.isSpectator() || this.isPassenger()) {
             this.setOnGround(false);
         }
