From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Initial Source <auto@mated.null>
Date: Wed, 7 Jun 2023 18:49:45 +1000
Subject: [PATCH] Initial


diff --git a/src/main/java/net/minecraft/world/level/block/AbstractCauldronBlock.java b/src/main/java/net/minecraft/world/level/block/AbstractCauldronBlock.java
new file mode 100644
index 0000000000000000000000000000000000000000..47468086c1cae252aa99c55b0065f225357dee62
--- /dev/null
+++ b/src/main/java/net/minecraft/world/level/block/AbstractCauldronBlock.java
@@ -0,0 +1,97 @@
+package net.minecraft.world.level.block;
+
+import java.util.Map;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.cauldron.CauldronInteraction;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.util.RandomSource;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResult;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.BlockGetter;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.state.BlockBehaviour;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.material.Fluid;
+import net.minecraft.world.level.material.Fluids;
+import net.minecraft.world.level.pathfinder.PathComputationType;
+import net.minecraft.world.phys.BlockHitResult;
+import net.minecraft.world.phys.shapes.BooleanOp;
+import net.minecraft.world.phys.shapes.CollisionContext;
+import net.minecraft.world.phys.shapes.Shapes;
+import net.minecraft.world.phys.shapes.VoxelShape;
+
+public abstract class AbstractCauldronBlock extends Block {
+    private static final int SIDE_THICKNESS = 2;
+    private static final int LEG_WIDTH = 4;
+    private static final int LEG_HEIGHT = 3;
+    private static final int LEG_DEPTH = 2;
+    protected static final int FLOOR_LEVEL = 4;
+    private static final VoxelShape INSIDE = box(2.0D, 4.0D, 2.0D, 14.0D, 16.0D, 14.0D);
+    protected static final VoxelShape SHAPE = Shapes.join(Shapes.block(), Shapes.or(box(0.0D, 0.0D, 4.0D, 16.0D, 3.0D, 12.0D), box(4.0D, 0.0D, 0.0D, 12.0D, 3.0D, 16.0D), box(2.0D, 0.0D, 2.0D, 14.0D, 3.0D, 14.0D), INSIDE), BooleanOp.ONLY_FIRST);
+    private final Map<Item, CauldronInteraction> interactions;
+
+    public AbstractCauldronBlock(BlockBehaviour.Properties settings, Map<Item, CauldronInteraction> behaviorMap) {
+        super(settings);
+        this.interactions = behaviorMap;
+    }
+
+    protected double getContentHeight(BlockState state) {
+        return 0.0D;
+    }
+
+    protected boolean isEntityInsideContent(BlockState state, BlockPos pos, Entity entity) {
+        return entity.getY() < (double)pos.getY() + this.getContentHeight(state) && entity.getBoundingBox().maxY > (double)pos.getY() + 0.25D;
+    }
+
+    @Override
+    public InteractionResult use(BlockState state, Level world, BlockPos pos, Player player, InteractionHand hand, BlockHitResult hit) {
+        ItemStack itemStack = player.getItemInHand(hand);
+        CauldronInteraction cauldronInteraction = this.interactions.get(itemStack.getItem());
+        return cauldronInteraction.interact(state, world, pos, player, hand, itemStack);
+    }
+
+    @Override
+    public VoxelShape getShape(BlockState state, BlockGetter world, BlockPos pos, CollisionContext context) {
+        return SHAPE;
+    }
+
+    @Override
+    public VoxelShape getInteractionShape(BlockState state, BlockGetter world, BlockPos pos) {
+        return INSIDE;
+    }
+
+    @Override
+    public boolean hasAnalogOutputSignal(BlockState state) {
+        return true;
+    }
+
+    @Override
+    public boolean isPathfindable(BlockState state, BlockGetter world, BlockPos pos, PathComputationType type) {
+        return false;
+    }
+
+    public abstract boolean isFull(BlockState state);
+
+    @Override
+    public void tick(BlockState state, ServerLevel world, BlockPos pos, RandomSource random) {
+        BlockPos blockPos = PointedDripstoneBlock.findStalactiteTipAboveCauldron(world, pos);
+        if (blockPos != null) {
+            Fluid fluid = PointedDripstoneBlock.getCauldronFillFluidType(world, blockPos);
+            if (fluid != Fluids.EMPTY && this.canReceiveStalactiteDrip(fluid)) {
+                this.receiveStalactiteDrip(state, world, pos, fluid);
+            }
+
+        }
+    }
+
+    protected boolean canReceiveStalactiteDrip(Fluid fluid) {
+        return false;
+    }
+
+    protected void receiveStalactiteDrip(BlockState state, Level world, BlockPos pos, Fluid fluid) {
+    }
+}
diff --git a/src/main/java/net/minecraft/world/level/block/AnvilBlock.java b/src/main/java/net/minecraft/world/level/block/AnvilBlock.java
new file mode 100644
index 0000000000000000000000000000000000000000..5c5a3b169795bf8a527b316c666cbc2105c66622
--- /dev/null
+++ b/src/main/java/net/minecraft/world/level/block/AnvilBlock.java
@@ -0,0 +1,135 @@
+package net.minecraft.world.level.block;
+
+import javax.annotation.Nullable;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.network.chat.Component;
+import net.minecraft.stats.Stats;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResult;
+import net.minecraft.world.MenuProvider;
+import net.minecraft.world.SimpleMenuProvider;
+import net.minecraft.world.damagesource.DamageSource;
+import net.minecraft.world.entity.Entity;
+import net.minecraft.world.entity.item.FallingBlockEntity;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.inventory.AnvilMenu;
+import net.minecraft.world.inventory.ContainerLevelAccess;
+import net.minecraft.world.item.context.BlockPlaceContext;
+import net.minecraft.world.level.BlockGetter;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.state.BlockBehaviour;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.block.state.StateDefinition;
+import net.minecraft.world.level.block.state.properties.DirectionProperty;
+import net.minecraft.world.level.pathfinder.PathComputationType;
+import net.minecraft.world.phys.BlockHitResult;
+import net.minecraft.world.phys.shapes.CollisionContext;
+import net.minecraft.world.phys.shapes.Shapes;
+import net.minecraft.world.phys.shapes.VoxelShape;
+
+public class AnvilBlock extends FallingBlock {
+    public static final DirectionProperty FACING = HorizontalDirectionalBlock.FACING;
+    private static final VoxelShape BASE = Block.box(2.0D, 0.0D, 2.0D, 14.0D, 4.0D, 14.0D);
+    private static final VoxelShape X_LEG1 = Block.box(3.0D, 4.0D, 4.0D, 13.0D, 5.0D, 12.0D);
+    private static final VoxelShape X_LEG2 = Block.box(4.0D, 5.0D, 6.0D, 12.0D, 10.0D, 10.0D);
+    private static final VoxelShape X_TOP = Block.box(0.0D, 10.0D, 3.0D, 16.0D, 16.0D, 13.0D);
+    private static final VoxelShape Z_LEG1 = Block.box(4.0D, 4.0D, 3.0D, 12.0D, 5.0D, 13.0D);
+    private static final VoxelShape Z_LEG2 = Block.box(6.0D, 5.0D, 4.0D, 10.0D, 10.0D, 12.0D);
+    private static final VoxelShape Z_TOP = Block.box(3.0D, 10.0D, 0.0D, 13.0D, 16.0D, 16.0D);
+    private static final VoxelShape X_AXIS_AABB = Shapes.or(BASE, X_LEG1, X_LEG2, X_TOP);
+    private static final VoxelShape Z_AXIS_AABB = Shapes.or(BASE, Z_LEG1, Z_LEG2, Z_TOP);
+    private static final Component CONTAINER_TITLE = Component.translatable("container.repair");
+    private static final float FALL_DAMAGE_PER_DISTANCE = 2.0F;
+    private static final int FALL_DAMAGE_MAX = 40;
+
+    public AnvilBlock(BlockBehaviour.Properties settings) {
+        super(settings);
+        this.registerDefaultState(this.stateDefinition.any().setValue(FACING, Direction.NORTH));
+    }
+
+    @Override
+    public BlockState getStateForPlacement(BlockPlaceContext ctx) {
+        return this.defaultBlockState().setValue(FACING, ctx.getHorizontalDirection().getClockWise());
+    }
+
+    @Override
+    public InteractionResult use(BlockState state, Level world, BlockPos pos, Player player, InteractionHand hand, BlockHitResult hit) {
+        if (world.isClientSide) {
+            return InteractionResult.SUCCESS;
+        } else {
+            player.openMenu(state.getMenuProvider(world, pos));
+            player.awardStat(Stats.INTERACT_WITH_ANVIL);
+            return InteractionResult.CONSUME;
+        }
+    }
+
+    @Nullable
+    @Override
+    public MenuProvider getMenuProvider(BlockState state, Level world, BlockPos pos) {
+        return new SimpleMenuProvider((syncId, inventory, player) -> {
+            return new AnvilMenu(syncId, inventory, ContainerLevelAccess.create(world, pos));
+        }, CONTAINER_TITLE);
+    }
+
+    @Override
+    public VoxelShape getShape(BlockState state, BlockGetter world, BlockPos pos, CollisionContext context) {
+        Direction direction = state.getValue(FACING);
+        return direction.getAxis() == Direction.Axis.X ? X_AXIS_AABB : Z_AXIS_AABB;
+    }
+
+    @Override
+    protected void falling(FallingBlockEntity entity) {
+        entity.setHurtsEntities(2.0F, 40);
+    }
+
+    @Override
+    public void onLand(Level world, BlockPos pos, BlockState fallingBlockState, BlockState currentStateInPos, FallingBlockEntity fallingBlockEntity) {
+        if (!fallingBlockEntity.isSilent()) {
+            world.levelEvent(1031, pos, 0);
+        }
+
+    }
+
+    @Override
+    public void onBrokenAfterFall(Level world, BlockPos pos, FallingBlockEntity fallingBlockEntity) {
+        if (!fallingBlockEntity.isSilent()) {
+            world.levelEvent(1029, pos, 0);
+        }
+
+    }
+
+    @Override
+    public DamageSource getFallDamageSource(Entity attacker) {
+        return attacker.damageSources().anvil(attacker);
+    }
+
+    @Nullable
+    public static BlockState damage(BlockState fallingState) {
+        if (fallingState.is(Blocks.ANVIL)) {
+            return Blocks.CHIPPED_ANVIL.defaultBlockState().setValue(FACING, fallingState.getValue(FACING));
+        } else {
+            return fallingState.is(Blocks.CHIPPED_ANVIL) ? Blocks.DAMAGED_ANVIL.defaultBlockState().setValue(FACING, fallingState.getValue(FACING)) : null;
+        }
+    }
+
+    @Override
+    public BlockState rotate(BlockState state, Rotation rotation) {
+        return state.setValue(FACING, rotation.rotate(state.getValue(FACING)));
+    }
+
+    @Override
+    protected void createBlockStateDefinition(StateDefinition.Builder<Block, BlockState> builder) {
+        builder.add(FACING);
+    }
+
+    @Override
+    public boolean isPathfindable(BlockState state, BlockGetter world, BlockPos pos, PathComputationType type) {
+        return false;
+    }
+
+    @Override
+    public int getDustColor(BlockState state, BlockGetter world, BlockPos pos) {
+        return state.getMapColor(world, pos).col;
+    }
+}
diff --git a/src/main/java/net/minecraft/world/level/block/BrewingStandBlock.java b/src/main/java/net/minecraft/world/level/block/BrewingStandBlock.java
new file mode 100644
index 0000000000000000000000000000000000000000..52c80b6bf4c07e39c247fd8400193ff6d73d1fbe
--- /dev/null
+++ b/src/main/java/net/minecraft/world/level/block/BrewingStandBlock.java
@@ -0,0 +1,127 @@
+package net.minecraft.world.level.block;
+
+import javax.annotation.Nullable;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.particles.ParticleTypes;
+import net.minecraft.stats.Stats;
+import net.minecraft.util.RandomSource;
+import net.minecraft.world.Containers;
+import net.minecraft.world.InteractionHand;
+import net.minecraft.world.InteractionResult;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.inventory.AbstractContainerMenu;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.BlockGetter;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.entity.BlockEntity;
+import net.minecraft.world.level.block.entity.BlockEntityTicker;
+import net.minecraft.world.level.block.entity.BlockEntityType;
+import net.minecraft.world.level.block.entity.BrewingStandBlockEntity;
+import net.minecraft.world.level.block.state.BlockBehaviour;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.block.state.StateDefinition;
+import net.minecraft.world.level.block.state.properties.BlockStateProperties;
+import net.minecraft.world.level.block.state.properties.BooleanProperty;
+import net.minecraft.world.level.pathfinder.PathComputationType;
+import net.minecraft.world.phys.BlockHitResult;
+import net.minecraft.world.phys.shapes.CollisionContext;
+import net.minecraft.world.phys.shapes.Shapes;
+import net.minecraft.world.phys.shapes.VoxelShape;
+
+public class BrewingStandBlock extends BaseEntityBlock {
+    public static final BooleanProperty[] HAS_BOTTLE = new BooleanProperty[]{BlockStateProperties.HAS_BOTTLE_0, BlockStateProperties.HAS_BOTTLE_1, BlockStateProperties.HAS_BOTTLE_2};
+    protected static final VoxelShape SHAPE = Shapes.or(Block.box(1.0D, 0.0D, 1.0D, 15.0D, 2.0D, 15.0D), Block.box(7.0D, 0.0D, 7.0D, 9.0D, 14.0D, 9.0D));
+
+    public BrewingStandBlock(BlockBehaviour.Properties settings) {
+        super(settings);
+        this.registerDefaultState(this.stateDefinition.any().setValue(HAS_BOTTLE[0], Boolean.valueOf(false)).setValue(HAS_BOTTLE[1], Boolean.valueOf(false)).setValue(HAS_BOTTLE[2], Boolean.valueOf(false)));
+    }
+
+    @Override
+    public RenderShape getRenderShape(BlockState state) {
+        return RenderShape.MODEL;
+    }
+
+    @Override
+    public BlockEntity newBlockEntity(BlockPos pos, BlockState state) {
+        return new BrewingStandBlockEntity(pos, state);
+    }
+
+    @Nullable
+    @Override
+    public <T extends BlockEntity> BlockEntityTicker<T> getTicker(Level world, BlockState state, BlockEntityType<T> type) {
+        return world.isClientSide ? null : createTickerHelper(type, BlockEntityType.BREWING_STAND, BrewingStandBlockEntity::serverTick);
+    }
+
+    @Override
+    public VoxelShape getShape(BlockState state, BlockGetter world, BlockPos pos, CollisionContext context) {
+        return SHAPE;
+    }
+
+    @Override
+    public InteractionResult use(BlockState state, Level world, BlockPos pos, Player player, InteractionHand hand, BlockHitResult hit) {
+        if (world.isClientSide) {
+            return InteractionResult.SUCCESS;
+        } else {
+            BlockEntity blockEntity = world.getBlockEntity(pos);
+            if (blockEntity instanceof BrewingStandBlockEntity) {
+                player.openMenu((BrewingStandBlockEntity)blockEntity);
+                player.awardStat(Stats.INTERACT_WITH_BREWINGSTAND);
+            }
+
+            return InteractionResult.CONSUME;
+        }
+    }
+
+    @Override
+    public void setPlacedBy(Level world, BlockPos pos, BlockState state, LivingEntity placer, ItemStack itemStack) {
+        if (itemStack.hasCustomHoverName()) {
+            BlockEntity blockEntity = world.getBlockEntity(pos);
+            if (blockEntity instanceof BrewingStandBlockEntity) {
+                ((BrewingStandBlockEntity)blockEntity).setCustomName(itemStack.getHoverName());
+            }
+        }
+
+    }
+
+    @Override
+    public void animateTick(BlockState state, Level world, BlockPos pos, RandomSource random) {
+        double d = (double)pos.getX() + 0.4D + (double)random.nextFloat() * 0.2D;
+        double e = (double)pos.getY() + 0.7D + (double)random.nextFloat() * 0.3D;
+        double f = (double)pos.getZ() + 0.4D + (double)random.nextFloat() * 0.2D;
+        world.addParticle(ParticleTypes.SMOKE, d, e, f, 0.0D, 0.0D, 0.0D);
+    }
+
+    @Override
+    public void onRemove(BlockState state, Level world, BlockPos pos, BlockState newState, boolean moved) {
+        if (!state.is(newState.getBlock())) {
+            BlockEntity blockEntity = world.getBlockEntity(pos);
+            if (blockEntity instanceof BrewingStandBlockEntity) {
+                Containers.dropContents(world, pos, (BrewingStandBlockEntity)blockEntity);
+            }
+
+            super.onRemove(state, world, pos, newState, moved);
+        }
+    }
+
+    @Override
+    public boolean hasAnalogOutputSignal(BlockState state) {
+        return true;
+    }
+
+    @Override
+    public int getAnalogOutputSignal(BlockState state, Level world, BlockPos pos) {
+        return AbstractContainerMenu.getRedstoneSignalFromBlockEntity(world.getBlockEntity(pos));
+    }
+
+    @Override
+    protected void createBlockStateDefinition(StateDefinition.Builder<Block, BlockState> builder) {
+        builder.add(HAS_BOTTLE[0], HAS_BOTTLE[1], HAS_BOTTLE[2]);
+    }
+
+    @Override
+    public boolean isPathfindable(BlockState state, BlockGetter world, BlockPos pos, PathComputationType type) {
+        return false;
+    }
+}
diff --git a/src/main/java/net/minecraft/world/level/block/EndPortalFrameBlock.java b/src/main/java/net/minecraft/world/level/block/EndPortalFrameBlock.java
new file mode 100644
index 0000000000000000000000000000000000000000..3f953bc9b6c9835f4dd5e81564f3d3cd85223c86
--- /dev/null
+++ b/src/main/java/net/minecraft/world/level/block/EndPortalFrameBlock.java
@@ -0,0 +1,89 @@
+package net.minecraft.world.level.block;
+
+import com.google.common.base.Predicates;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.world.item.context.BlockPlaceContext;
+import net.minecraft.world.level.BlockGetter;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.level.block.state.BlockBehaviour;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.block.state.StateDefinition;
+import net.minecraft.world.level.block.state.pattern.BlockInWorld;
+import net.minecraft.world.level.block.state.pattern.BlockPattern;
+import net.minecraft.world.level.block.state.pattern.BlockPatternBuilder;
+import net.minecraft.world.level.block.state.predicate.BlockStatePredicate;
+import net.minecraft.world.level.block.state.properties.BlockStateProperties;
+import net.minecraft.world.level.block.state.properties.BooleanProperty;
+import net.minecraft.world.level.block.state.properties.DirectionProperty;
+import net.minecraft.world.level.pathfinder.PathComputationType;
+import net.minecraft.world.phys.shapes.CollisionContext;
+import net.minecraft.world.phys.shapes.Shapes;
+import net.minecraft.world.phys.shapes.VoxelShape;
+
+public class EndPortalFrameBlock extends Block {
+    public static final DirectionProperty FACING = HorizontalDirectionalBlock.FACING;
+    public static final BooleanProperty HAS_EYE = BlockStateProperties.EYE;
+    protected static final VoxelShape BASE_SHAPE = Block.box(0.0D, 0.0D, 0.0D, 16.0D, 13.0D, 16.0D);
+    protected static final VoxelShape EYE_SHAPE = Block.box(4.0D, 13.0D, 4.0D, 12.0D, 16.0D, 12.0D);
+    protected static final VoxelShape FULL_SHAPE = Shapes.or(BASE_SHAPE, EYE_SHAPE);
+    private static BlockPattern portalShape;
+
+    public EndPortalFrameBlock(BlockBehaviour.Properties settings) {
+        super(settings);
+        this.registerDefaultState(this.stateDefinition.any().setValue(FACING, Direction.NORTH).setValue(HAS_EYE, Boolean.valueOf(false)));
+    }
+
+    @Override
+    public boolean useShapeForLightOcclusion(BlockState state) {
+        return true;
+    }
+
+    @Override
+    public VoxelShape getShape(BlockState state, BlockGetter world, BlockPos pos, CollisionContext context) {
+        return state.getValue(HAS_EYE) ? FULL_SHAPE : BASE_SHAPE;
+    }
+
+    @Override
+    public BlockState getStateForPlacement(BlockPlaceContext ctx) {
+        return this.defaultBlockState().setValue(FACING, ctx.getHorizontalDirection().getOpposite()).setValue(HAS_EYE, Boolean.valueOf(false));
+    }
+
+    @Override
+    public boolean hasAnalogOutputSignal(BlockState state) {
+        return true;
+    }
+
+    @Override
+    public int getAnalogOutputSignal(BlockState state, Level world, BlockPos pos) {
+        return state.getValue(HAS_EYE) ? 15 : 0;
+    }
+
+    @Override
+    public BlockState rotate(BlockState state, Rotation rotation) {
+        return state.setValue(FACING, rotation.rotate(state.getValue(FACING)));
+    }
+
+    @Override
+    public BlockState mirror(BlockState state, Mirror mirror) {
+        return state.rotate(mirror.getRotation(state.getValue(FACING)));
+    }
+
+    @Override
+    protected void createBlockStateDefinition(StateDefinition.Builder<Block, BlockState> builder) {
+        builder.add(FACING, HAS_EYE);
+    }
+
+    public static BlockPattern getOrCreatePortalShape() {
+        if (portalShape == null) {
+            portalShape = BlockPatternBuilder.start().aisle("?vvv?", ">???<", ">???<", ">???<", "?^^^?").where('?', BlockInWorld.hasState(BlockStatePredicate.ANY)).where('^', BlockInWorld.hasState(BlockStatePredicate.forBlock(Blocks.END_PORTAL_FRAME).where(HAS_EYE, Predicates.equalTo(true)).where(FACING, Predicates.equalTo(Direction.SOUTH)))).where('>', BlockInWorld.hasState(BlockStatePredicate.forBlock(Blocks.END_PORTAL_FRAME).where(HAS_EYE, Predicates.equalTo(true)).where(FACING, Predicates.equalTo(Direction.WEST)))).where('v', BlockInWorld.hasState(BlockStatePredicate.forBlock(Blocks.END_PORTAL_FRAME).where(HAS_EYE, Predicates.equalTo(true)).where(FACING, Predicates.equalTo(Direction.NORTH)))).where('<', BlockInWorld.hasState(BlockStatePredicate.forBlock(Blocks.END_PORTAL_FRAME).where(HAS_EYE, Predicates.equalTo(true)).where(FACING, Predicates.equalTo(Direction.EAST)))).build();
+        }
+
+        return portalShape;
+    }
+
+    @Override
+    public boolean isPathfindable(BlockState state, BlockGetter world, BlockPos pos, PathComputationType type) {
+        return false;
+    }
+}
diff --git a/src/main/java/net/minecraft/world/level/block/IronBarsBlock.java b/src/main/java/net/minecraft/world/level/block/IronBarsBlock.java
new file mode 100644
index 0000000000000000000000000000000000000000..83352a3909e892e35f96adea092a8c399ad98463
--- /dev/null
+++ b/src/main/java/net/minecraft/world/level/block/IronBarsBlock.java
@@ -0,0 +1,77 @@
+package net.minecraft.world.level.block;
+
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.tags.BlockTags;
+import net.minecraft.world.item.context.BlockPlaceContext;
+import net.minecraft.world.level.BlockGetter;
+import net.minecraft.world.level.LevelAccessor;
+import net.minecraft.world.level.block.state.BlockBehaviour;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.block.state.StateDefinition;
+import net.minecraft.world.level.material.FluidState;
+import net.minecraft.world.level.material.Fluids;
+import net.minecraft.world.phys.shapes.CollisionContext;
+import net.minecraft.world.phys.shapes.Shapes;
+import net.minecraft.world.phys.shapes.VoxelShape;
+
+public class IronBarsBlock extends CrossCollisionBlock {
+    protected IronBarsBlock(BlockBehaviour.Properties settings) {
+        super(1.0F, 1.0F, 16.0F, 16.0F, 16.0F, settings);
+        this.registerDefaultState(this.stateDefinition.any().setValue(NORTH, Boolean.valueOf(false)).setValue(EAST, Boolean.valueOf(false)).setValue(SOUTH, Boolean.valueOf(false)).setValue(WEST, Boolean.valueOf(false)).setValue(WATERLOGGED, Boolean.valueOf(false)));
+    }
+
+    @Override
+    public BlockState getStateForPlacement(BlockPlaceContext ctx) {
+        BlockGetter blockGetter = ctx.getLevel();
+        BlockPos blockPos = ctx.getClickedPos();
+        FluidState fluidState = ctx.getLevel().getFluidState(ctx.getClickedPos());
+        BlockPos blockPos2 = blockPos.north();
+        BlockPos blockPos3 = blockPos.south();
+        BlockPos blockPos4 = blockPos.west();
+        BlockPos blockPos5 = blockPos.east();
+        BlockState blockState = blockGetter.getBlockState(blockPos2);
+        BlockState blockState2 = blockGetter.getBlockState(blockPos3);
+        BlockState blockState3 = blockGetter.getBlockState(blockPos4);
+        BlockState blockState4 = blockGetter.getBlockState(blockPos5);
+        return this.defaultBlockState().setValue(NORTH, Boolean.valueOf(this.attachsTo(blockState, blockState.isFaceSturdy(blockGetter, blockPos2, Direction.SOUTH)))).setValue(SOUTH, Boolean.valueOf(this.attachsTo(blockState2, blockState2.isFaceSturdy(blockGetter, blockPos3, Direction.NORTH)))).setValue(WEST, Boolean.valueOf(this.attachsTo(blockState3, blockState3.isFaceSturdy(blockGetter, blockPos4, Direction.EAST)))).setValue(EAST, Boolean.valueOf(this.attachsTo(blockState4, blockState4.isFaceSturdy(blockGetter, blockPos5, Direction.WEST)))).setValue(WATERLOGGED, Boolean.valueOf(fluidState.getType() == Fluids.WATER));
+    }
+
+    @Override
+    public BlockState updateShape(BlockState state, Direction direction, BlockState neighborState, LevelAccessor world, BlockPos pos, BlockPos neighborPos) {
+        if (state.getValue(WATERLOGGED)) {
+            world.scheduleTick(pos, Fluids.WATER, Fluids.WATER.getTickDelay(world));
+        }
+
+        return direction.getAxis().isHorizontal() ? state.setValue(PROPERTY_BY_DIRECTION.get(direction), Boolean.valueOf(this.attachsTo(neighborState, neighborState.isFaceSturdy(world, neighborPos, direction.getOpposite())))) : super.updateShape(state, direction, neighborState, world, pos, neighborPos);
+    }
+
+    @Override
+    public VoxelShape getVisualShape(BlockState state, BlockGetter world, BlockPos pos, CollisionContext context) {
+        return Shapes.empty();
+    }
+
+    @Override
+    public boolean skipRendering(BlockState state, BlockState stateFrom, Direction direction) {
+        if (stateFrom.is(this)) {
+            if (!direction.getAxis().isHorizontal()) {
+                return true;
+            }
+
+            if (state.getValue(PROPERTY_BY_DIRECTION.get(direction)) && stateFrom.getValue(PROPERTY_BY_DIRECTION.get(direction.getOpposite()))) {
+                return true;
+            }
+        }
+
+        return super.skipRendering(state, stateFrom, direction);
+    }
+
+    public final boolean attachsTo(BlockState state, boolean sideSolidFullSquare) {
+        return !isExceptionForConnection(state) && sideSolidFullSquare || state.getBlock() instanceof IronBarsBlock || state.is(BlockTags.WALLS);
+    }
+
+    @Override
+    protected void createBlockStateDefinition(StateDefinition.Builder<Block, BlockState> builder) {
+        builder.add(NORTH, EAST, WEST, SOUTH, WATERLOGGED);
+    }
+}
diff --git a/src/main/java/net/minecraft/world/level/block/WallBlock.java b/src/main/java/net/minecraft/world/level/block/WallBlock.java
new file mode 100644
index 0000000000000000000000000000000000000000..d1d0fa66aa0c3e481b6d68fe36147040a730293f
--- /dev/null
+++ b/src/main/java/net/minecraft/world/level/block/WallBlock.java
@@ -0,0 +1,275 @@
+package net.minecraft.world.level.block;
+
+import com.google.common.collect.ImmutableMap;
+import java.util.Map;
+import net.minecraft.core.BlockPos;
+import net.minecraft.core.Direction;
+import net.minecraft.tags.BlockTags;
+import net.minecraft.world.item.context.BlockPlaceContext;
+import net.minecraft.world.level.BlockGetter;
+import net.minecraft.world.level.LevelAccessor;
+import net.minecraft.world.level.LevelReader;
+import net.minecraft.world.level.block.state.BlockBehaviour;
+import net.minecraft.world.level.block.state.BlockState;
+import net.minecraft.world.level.block.state.StateDefinition;
+import net.minecraft.world.level.block.state.properties.BlockStateProperties;
+import net.minecraft.world.level.block.state.properties.BooleanProperty;
+import net.minecraft.world.level.block.state.properties.EnumProperty;
+import net.minecraft.world.level.block.state.properties.Property;
+import net.minecraft.world.level.block.state.properties.WallSide;
+import net.minecraft.world.level.material.FluidState;
+import net.minecraft.world.level.material.Fluids;
+import net.minecraft.world.level.pathfinder.PathComputationType;
+import net.minecraft.world.phys.shapes.BooleanOp;
+import net.minecraft.world.phys.shapes.CollisionContext;
+import net.minecraft.world.phys.shapes.Shapes;
+import net.minecraft.world.phys.shapes.VoxelShape;
+
+public class WallBlock extends Block implements SimpleWaterloggedBlock {
+    public static final BooleanProperty UP = BlockStateProperties.UP;
+    public static final EnumProperty<WallSide> EAST_WALL = BlockStateProperties.EAST_WALL;
+    public static final EnumProperty<WallSide> NORTH_WALL = BlockStateProperties.NORTH_WALL;
+    public static final EnumProperty<WallSide> SOUTH_WALL = BlockStateProperties.SOUTH_WALL;
+    public static final EnumProperty<WallSide> WEST_WALL = BlockStateProperties.WEST_WALL;
+    public static final BooleanProperty WATERLOGGED = BlockStateProperties.WATERLOGGED;
+    private final Map<BlockState, VoxelShape> shapeByIndex;
+    private final Map<BlockState, VoxelShape> collisionShapeByIndex;
+    private static final int WALL_WIDTH = 3;
+    private static final int WALL_HEIGHT = 14;
+    private static final int POST_WIDTH = 4;
+    private static final int POST_COVER_WIDTH = 1;
+    private static final int WALL_COVER_START = 7;
+    private static final int WALL_COVER_END = 9;
+    private static final VoxelShape POST_TEST = Block.box(7.0D, 0.0D, 7.0D, 9.0D, 16.0D, 9.0D);
+    private static final VoxelShape NORTH_TEST = Block.box(7.0D, 0.0D, 0.0D, 9.0D, 16.0D, 9.0D);
+    private static final VoxelShape SOUTH_TEST = Block.box(7.0D, 0.0D, 7.0D, 9.0D, 16.0D, 16.0D);
+    private static final VoxelShape WEST_TEST = Block.box(0.0D, 0.0D, 7.0D, 9.0D, 16.0D, 9.0D);
+    private static final VoxelShape EAST_TEST = Block.box(7.0D, 0.0D, 7.0D, 16.0D, 16.0D, 9.0D);
+
+    public WallBlock(BlockBehaviour.Properties settings) {
+        super(settings);
+        this.registerDefaultState(this.stateDefinition.any().setValue(UP, Boolean.valueOf(true)).setValue(NORTH_WALL, WallSide.NONE).setValue(EAST_WALL, WallSide.NONE).setValue(SOUTH_WALL, WallSide.NONE).setValue(WEST_WALL, WallSide.NONE).setValue(WATERLOGGED, Boolean.valueOf(false)));
+        this.shapeByIndex = this.makeShapes(4.0F, 3.0F, 16.0F, 0.0F, 14.0F, 16.0F);
+        this.collisionShapeByIndex = this.makeShapes(4.0F, 3.0F, 24.0F, 0.0F, 24.0F, 24.0F);
+    }
+
+    private static VoxelShape applyWallShape(VoxelShape base, WallSide wallShape, VoxelShape tall, VoxelShape low) {
+        if (wallShape == WallSide.TALL) {
+            return Shapes.or(base, low);
+        } else {
+            return wallShape == WallSide.LOW ? Shapes.or(base, tall) : base;
+        }
+    }
+
+    private Map<BlockState, VoxelShape> makeShapes(float f, float g, float h, float i, float j, float k) {
+        float l = 8.0F - f;
+        float m = 8.0F + f;
+        float n = 8.0F - g;
+        float o = 8.0F + g;
+        VoxelShape voxelShape = Block.box((double)l, 0.0D, (double)l, (double)m, (double)h, (double)m);
+        VoxelShape voxelShape2 = Block.box((double)n, (double)i, 0.0D, (double)o, (double)j, (double)o);
+        VoxelShape voxelShape3 = Block.box((double)n, (double)i, (double)n, (double)o, (double)j, 16.0D);
+        VoxelShape voxelShape4 = Block.box(0.0D, (double)i, (double)n, (double)o, (double)j, (double)o);
+        VoxelShape voxelShape5 = Block.box((double)n, (double)i, (double)n, 16.0D, (double)j, (double)o);
+        VoxelShape voxelShape6 = Block.box((double)n, (double)i, 0.0D, (double)o, (double)k, (double)o);
+        VoxelShape voxelShape7 = Block.box((double)n, (double)i, (double)n, (double)o, (double)k, 16.0D);
+        VoxelShape voxelShape8 = Block.box(0.0D, (double)i, (double)n, (double)o, (double)k, (double)o);
+        VoxelShape voxelShape9 = Block.box((double)n, (double)i, (double)n, 16.0D, (double)k, (double)o);
+        ImmutableMap.Builder<BlockState, VoxelShape> builder = ImmutableMap.builder();
+
+        for(Boolean boolean_ : UP.getPossibleValues()) {
+            for(WallSide wallSide : EAST_WALL.getPossibleValues()) {
+                for(WallSide wallSide2 : NORTH_WALL.getPossibleValues()) {
+                    for(WallSide wallSide3 : WEST_WALL.getPossibleValues()) {
+                        for(WallSide wallSide4 : SOUTH_WALL.getPossibleValues()) {
+                            VoxelShape voxelShape10 = Shapes.empty();
+                            voxelShape10 = applyWallShape(voxelShape10, wallSide, voxelShape5, voxelShape9);
+                            voxelShape10 = applyWallShape(voxelShape10, wallSide3, voxelShape4, voxelShape8);
+                            voxelShape10 = applyWallShape(voxelShape10, wallSide2, voxelShape2, voxelShape6);
+                            voxelShape10 = applyWallShape(voxelShape10, wallSide4, voxelShape3, voxelShape7);
+                            if (boolean_) {
+                                voxelShape10 = Shapes.or(voxelShape10, voxelShape);
+                            }
+
+                            BlockState blockState = this.defaultBlockState().setValue(UP, boolean_).setValue(EAST_WALL, wallSide).setValue(WEST_WALL, wallSide3).setValue(NORTH_WALL, wallSide2).setValue(SOUTH_WALL, wallSide4);
+                            builder.put(blockState.setValue(WATERLOGGED, Boolean.valueOf(false)), voxelShape10);
+                            builder.put(blockState.setValue(WATERLOGGED, Boolean.valueOf(true)), voxelShape10);
+                        }
+                    }
+                }
+            }
+        }
+
+        return builder.build();
+    }
+
+    @Override
+    public VoxelShape getShape(BlockState state, BlockGetter world, BlockPos pos, CollisionContext context) {
+        return this.shapeByIndex.get(state);
+    }
+
+    @Override
+    public VoxelShape getCollisionShape(BlockState state, BlockGetter world, BlockPos pos, CollisionContext context) {
+        return this.collisionShapeByIndex.get(state);
+    }
+
+    @Override
+    public boolean isPathfindable(BlockState state, BlockGetter world, BlockPos pos, PathComputationType type) {
+        return false;
+    }
+
+    private boolean connectsTo(BlockState state, boolean faceFullSquare, Direction side) {
+        Block block = state.getBlock();
+        boolean bl = block instanceof FenceGateBlock && FenceGateBlock.connectsToDirection(state, side);
+        return state.is(BlockTags.WALLS) || !isExceptionForConnection(state) && faceFullSquare || block instanceof IronBarsBlock || bl;
+    }
+
+    @Override
+    public BlockState getStateForPlacement(BlockPlaceContext ctx) {
+        LevelReader levelReader = ctx.getLevel();
+        BlockPos blockPos = ctx.getClickedPos();
+        FluidState fluidState = ctx.getLevel().getFluidState(ctx.getClickedPos());
+        BlockPos blockPos2 = blockPos.north();
+        BlockPos blockPos3 = blockPos.east();
+        BlockPos blockPos4 = blockPos.south();
+        BlockPos blockPos5 = blockPos.west();
+        BlockPos blockPos6 = blockPos.above();
+        BlockState blockState = levelReader.getBlockState(blockPos2);
+        BlockState blockState2 = levelReader.getBlockState(blockPos3);
+        BlockState blockState3 = levelReader.getBlockState(blockPos4);
+        BlockState blockState4 = levelReader.getBlockState(blockPos5);
+        BlockState blockState5 = levelReader.getBlockState(blockPos6);
+        boolean bl = this.connectsTo(blockState, blockState.isFaceSturdy(levelReader, blockPos2, Direction.SOUTH), Direction.SOUTH);
+        boolean bl2 = this.connectsTo(blockState2, blockState2.isFaceSturdy(levelReader, blockPos3, Direction.WEST), Direction.WEST);
+        boolean bl3 = this.connectsTo(blockState3, blockState3.isFaceSturdy(levelReader, blockPos4, Direction.NORTH), Direction.NORTH);
+        boolean bl4 = this.connectsTo(blockState4, blockState4.isFaceSturdy(levelReader, blockPos5, Direction.EAST), Direction.EAST);
+        BlockState blockState6 = this.defaultBlockState().setValue(WATERLOGGED, Boolean.valueOf(fluidState.getType() == Fluids.WATER));
+        return this.updateShape(levelReader, blockState6, blockPos6, blockState5, bl, bl2, bl3, bl4);
+    }
+
+    @Override
+    public BlockState updateShape(BlockState state, Direction direction, BlockState neighborState, LevelAccessor world, BlockPos pos, BlockPos neighborPos) {
+        if (state.getValue(WATERLOGGED)) {
+            world.scheduleTick(pos, Fluids.WATER, Fluids.WATER.getTickDelay(world));
+        }
+
+        if (direction == Direction.DOWN) {
+            return super.updateShape(state, direction, neighborState, world, pos, neighborPos);
+        } else {
+            return direction == Direction.UP ? this.topUpdate(world, state, neighborPos, neighborState) : this.sideUpdate(world, pos, state, neighborPos, neighborState, direction);
+        }
+    }
+
+    private static boolean isConnected(BlockState state, Property<WallSide> property) {
+        return state.getValue(property) != WallSide.NONE;
+    }
+
+    private static boolean isCovered(VoxelShape aboveShape, VoxelShape tallShape) {
+        return !Shapes.joinIsNotEmpty(tallShape, aboveShape, BooleanOp.ONLY_FIRST);
+    }
+
+    private BlockState topUpdate(LevelReader world, BlockState state, BlockPos pos, BlockState aboveState) {
+        boolean bl = isConnected(state, NORTH_WALL);
+        boolean bl2 = isConnected(state, EAST_WALL);
+        boolean bl3 = isConnected(state, SOUTH_WALL);
+        boolean bl4 = isConnected(state, WEST_WALL);
+        return this.updateShape(world, state, pos, aboveState, bl, bl2, bl3, bl4);
+    }
+
+    private BlockState sideUpdate(LevelReader world, BlockPos pos, BlockState state, BlockPos neighborPos, BlockState neighborState, Direction direction) {
+        Direction direction2 = direction.getOpposite();
+        boolean bl = direction == Direction.NORTH ? this.connectsTo(neighborState, neighborState.isFaceSturdy(world, neighborPos, direction2), direction2) : isConnected(state, NORTH_WALL);
+        boolean bl2 = direction == Direction.EAST ? this.connectsTo(neighborState, neighborState.isFaceSturdy(world, neighborPos, direction2), direction2) : isConnected(state, EAST_WALL);
+        boolean bl3 = direction == Direction.SOUTH ? this.connectsTo(neighborState, neighborState.isFaceSturdy(world, neighborPos, direction2), direction2) : isConnected(state, SOUTH_WALL);
+        boolean bl4 = direction == Direction.WEST ? this.connectsTo(neighborState, neighborState.isFaceSturdy(world, neighborPos, direction2), direction2) : isConnected(state, WEST_WALL);
+        BlockPos blockPos = pos.above();
+        BlockState blockState = world.getBlockState(blockPos);
+        return this.updateShape(world, state, blockPos, blockState, bl, bl2, bl3, bl4);
+    }
+
+    private BlockState updateShape(LevelReader world, BlockState state, BlockPos pos, BlockState aboveState, boolean north, boolean east, boolean south, boolean west) {
+        VoxelShape voxelShape = aboveState.getCollisionShape(world, pos).getFaceShape(Direction.DOWN);
+        BlockState blockState = this.updateSides(state, north, east, south, west, voxelShape);
+        return blockState.setValue(UP, Boolean.valueOf(this.shouldRaisePost(blockState, aboveState, voxelShape)));
+    }
+
+    private boolean shouldRaisePost(BlockState state, BlockState aboveState, VoxelShape aboveShape) {
+        boolean bl = aboveState.getBlock() instanceof WallBlock && aboveState.getValue(UP);
+        if (bl) {
+            return true;
+        } else {
+            WallSide wallSide = state.getValue(NORTH_WALL);
+            WallSide wallSide2 = state.getValue(SOUTH_WALL);
+            WallSide wallSide3 = state.getValue(EAST_WALL);
+            WallSide wallSide4 = state.getValue(WEST_WALL);
+            boolean bl2 = wallSide2 == WallSide.NONE;
+            boolean bl3 = wallSide4 == WallSide.NONE;
+            boolean bl4 = wallSide3 == WallSide.NONE;
+            boolean bl5 = wallSide == WallSide.NONE;
+            boolean bl6 = bl5 && bl2 && bl3 && bl4 || bl5 != bl2 || bl3 != bl4;
+            if (bl6) {
+                return true;
+            } else {
+                boolean bl7 = wallSide == WallSide.TALL && wallSide2 == WallSide.TALL || wallSide3 == WallSide.TALL && wallSide4 == WallSide.TALL;
+                if (bl7) {
+                    return false;
+                } else {
+                    return aboveState.is(BlockTags.WALL_POST_OVERRIDE) || isCovered(aboveShape, POST_TEST);
+                }
+            }
+        }
+    }
+
+    private BlockState updateSides(BlockState state, boolean north, boolean east, boolean south, boolean west, VoxelShape aboveShape) {
+        return state.setValue(NORTH_WALL, this.makeWallState(north, aboveShape, NORTH_TEST)).setValue(EAST_WALL, this.makeWallState(east, aboveShape, EAST_TEST)).setValue(SOUTH_WALL, this.makeWallState(south, aboveShape, SOUTH_TEST)).setValue(WEST_WALL, this.makeWallState(west, aboveShape, WEST_TEST));
+    }
+
+    private WallSide makeWallState(boolean connected, VoxelShape aboveShape, VoxelShape tallShape) {
+        if (connected) {
+            return isCovered(aboveShape, tallShape) ? WallSide.TALL : WallSide.LOW;
+        } else {
+            return WallSide.NONE;
+        }
+    }
+
+    @Override
+    public FluidState getFluidState(BlockState state) {
+        return state.getValue(WATERLOGGED) ? Fluids.WATER.getSource(false) : super.getFluidState(state);
+    }
+
+    @Override
+    public boolean propagatesSkylightDown(BlockState state, BlockGetter world, BlockPos pos) {
+        return !state.getValue(WATERLOGGED);
+    }
+
+    @Override
+    protected void createBlockStateDefinition(StateDefinition.Builder<Block, BlockState> builder) {
+        builder.add(UP, NORTH_WALL, EAST_WALL, WEST_WALL, SOUTH_WALL, WATERLOGGED);
+    }
+
+    @Override
+    public BlockState rotate(BlockState state, Rotation rotation) {
+        switch (rotation) {
+            case CLOCKWISE_180:
+                return state.setValue(NORTH_WALL, state.getValue(SOUTH_WALL)).setValue(EAST_WALL, state.getValue(WEST_WALL)).setValue(SOUTH_WALL, state.getValue(NORTH_WALL)).setValue(WEST_WALL, state.getValue(EAST_WALL));
+            case COUNTERCLOCKWISE_90:
+                return state.setValue(NORTH_WALL, state.getValue(EAST_WALL)).setValue(EAST_WALL, state.getValue(SOUTH_WALL)).setValue(SOUTH_WALL, state.getValue(WEST_WALL)).setValue(WEST_WALL, state.getValue(NORTH_WALL));
+            case CLOCKWISE_90:
+                return state.setValue(NORTH_WALL, state.getValue(WEST_WALL)).setValue(EAST_WALL, state.getValue(NORTH_WALL)).setValue(SOUTH_WALL, state.getValue(EAST_WALL)).setValue(WEST_WALL, state.getValue(SOUTH_WALL));
+            default:
+                return state;
+        }
+    }
+
+    @Override
+    public BlockState mirror(BlockState state, Mirror mirror) {
+        switch (mirror) {
+            case LEFT_RIGHT:
+                return state.setValue(NORTH_WALL, state.getValue(SOUTH_WALL)).setValue(SOUTH_WALL, state.getValue(NORTH_WALL));
+            case FRONT_BACK:
+                return state.setValue(EAST_WALL, state.getValue(WEST_WALL)).setValue(WEST_WALL, state.getValue(EAST_WALL));
+            default:
+                return super.mirror(state, mirror);
+        }
+    }
+}
diff --git a/src/main/java/net/minecraft/world/level/block/entity/Hopper.java b/src/main/java/net/minecraft/world/level/block/entity/Hopper.java
new file mode 100644
index 0000000000000000000000000000000000000000..a05acf709735b40ca86f978508c63a86065fd405
--- /dev/null
+++ b/src/main/java/net/minecraft/world/level/block/entity/Hopper.java
@@ -0,0 +1,22 @@
+package net.minecraft.world.level.block.entity;
+
+import net.minecraft.world.Container;
+import net.minecraft.world.level.block.Block;
+import net.minecraft.world.phys.shapes.Shapes;
+import net.minecraft.world.phys.shapes.VoxelShape;
+
+public interface Hopper extends Container {
+    VoxelShape INSIDE = Block.box(2.0D, 11.0D, 2.0D, 14.0D, 16.0D, 14.0D);
+    VoxelShape ABOVE = Block.box(0.0D, 16.0D, 0.0D, 16.0D, 32.0D, 16.0D);
+    VoxelShape SUCK = Shapes.or(INSIDE, ABOVE);
+
+    default VoxelShape getSuckShape() {
+        return SUCK;
+    }
+
+    double getLevelX();
+
+    double getLevelY();
+
+    double getLevelZ();
+}
