From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mason Hunt <masonhunt130ii@gmail.com>
Date: Wed, 7 Jun 2023 19:13:55 +1000
Subject: [PATCH] 1.12 Bed Hitboxes


diff --git a/src/main/java/net/minecraft/world/level/block/BedBlock.java b/src/main/java/net/minecraft/world/level/block/BedBlock.java
index d1d5363ab1742add8ff45507a303106f4d65f52f..f4cf205088998161ead2ce6d0ceac59942627355 100644
--- a/src/main/java/net/minecraft/world/level/block/BedBlock.java
+++ b/src/main/java/net/minecraft/world/level/block/BedBlock.java
@@ -47,16 +47,7 @@ public class BedBlock extends HorizontalDirectionalBlock implements EntityBlock
     public static final EnumProperty<BedPart> PART = BlockStateProperties.BED_PART;
     public static final BooleanProperty OCCUPIED = BlockStateProperties.OCCUPIED;
     protected static final int HEIGHT = 9;
-    protected static final VoxelShape BASE = Block.box(0.0D, 3.0D, 0.0D, 16.0D, 9.0D, 16.0D);
-    private static final int LEG_WIDTH = 3;
-    protected static final VoxelShape LEG_NORTH_WEST = Block.box(0.0D, 0.0D, 0.0D, 3.0D, 3.0D, 3.0D);
-    protected static final VoxelShape LEG_SOUTH_WEST = Block.box(0.0D, 0.0D, 13.0D, 3.0D, 3.0D, 16.0D);
-    protected static final VoxelShape LEG_NORTH_EAST = Block.box(13.0D, 0.0D, 0.0D, 16.0D, 3.0D, 3.0D);
-    protected static final VoxelShape LEG_SOUTH_EAST = Block.box(13.0D, 0.0D, 13.0D, 16.0D, 3.0D, 16.0D);
-    protected static final VoxelShape NORTH_SHAPE = Shapes.or(BedBlock.BASE, BedBlock.LEG_NORTH_WEST, BedBlock.LEG_NORTH_EAST);
-    protected static final VoxelShape SOUTH_SHAPE = Shapes.or(BedBlock.BASE, BedBlock.LEG_SOUTH_WEST, BedBlock.LEG_SOUTH_EAST);
-    protected static final VoxelShape WEST_SHAPE = Shapes.or(BedBlock.BASE, BedBlock.LEG_NORTH_WEST, BedBlock.LEG_SOUTH_WEST);
-    protected static final VoxelShape EAST_SHAPE = Shapes.or(BedBlock.BASE, BedBlock.LEG_NORTH_EAST, BedBlock.LEG_SOUTH_EAST);
+    protected static final VoxelShape BASE = Block.box(0.0D, 0.0D, 0.0D, 16.0D, 9.0D, 16.0D);
     private final DyeColor color;
 
     public BedBlock(DyeColor color, BlockBehaviour.Properties settings) {
@@ -239,18 +230,7 @@ public class BedBlock extends HorizontalDirectionalBlock implements EntityBlock
 
     @Override
     public VoxelShape getShape(BlockState state, BlockGetter world, BlockPos pos, CollisionContext context) {
-        Direction enumdirection = BedBlock.getConnectedDirection(state).getOpposite();
-
-        switch (enumdirection) {
-            case NORTH:
-                return BedBlock.NORTH_SHAPE;
-            case SOUTH:
-                return BedBlock.SOUTH_SHAPE;
-            case WEST:
-                return BedBlock.WEST_SHAPE;
-            default:
-                return BedBlock.EAST_SHAPE;
-        }
+        return BASE;
     }
 
     public static Direction getConnectedDirection(BlockState state) {
