From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mason Hunt <masonhunt130ii@gmail.com>
Date: Wed, 7 Jun 2023 19:14:04 +1000
Subject: [PATCH] 1.12 Brewing Stand Hitboxes


diff --git a/src/main/java/net/minecraft/world/level/block/BrewingStandBlock.java b/src/main/java/net/minecraft/world/level/block/BrewingStandBlock.java
index 52c80b6bf4c07e39c247fd8400193ff6d73d1fbe..05279d4c961e84d633391114cd9a85e5dd277531 100644
--- a/src/main/java/net/minecraft/world/level/block/BrewingStandBlock.java
+++ b/src/main/java/net/minecraft/world/level/block/BrewingStandBlock.java
@@ -31,7 +31,7 @@ import net.minecraft.world.phys.shapes.VoxelShape;
 
 public class BrewingStandBlock extends BaseEntityBlock {
     public static final BooleanProperty[] HAS_BOTTLE = new BooleanProperty[]{BlockStateProperties.HAS_BOTTLE_0, BlockStateProperties.HAS_BOTTLE_1, BlockStateProperties.HAS_BOTTLE_2};
-    protected static final VoxelShape SHAPE = Shapes.or(Block.box(1.0D, 0.0D, 1.0D, 15.0D, 2.0D, 15.0D), Block.box(7.0D, 0.0D, 7.0D, 9.0D, 14.0D, 9.0D));
+    protected static final VoxelShape SHAPE = Shapes.or(Block.box(0.0D, 0.0D, 0.0D, 16.0D, 2.0D, 16.0D), Block.box(7.0D, 0.0D, 7.0D, 9.0D, 14.0D, 9.0D));
 
     public BrewingStandBlock(BlockBehaviour.Properties settings) {
         super(settings);
